lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -410,4 +417,102 @@ private List<String> extractTxnIdsFromCredentials(List<SignedJWT> credentials)
+                            : matchOperationalProfile(credentials, requestedVot);
+
+            if (attainedVotAndProfile.isPresent()) {
+                LOGGER.info(
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "Attained vot with profile")
+                                .with(
+                                        LOG_VOT.getFieldName(),
+                                        attainedVotAndProfile.get().attainedVot())
+                                .with(
+                                        LOG_PROFILE.getFieldName(),
+                                        attainedVotAndProfile.get().profileName()));
+
+                return attainedVotAndProfile;
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<VotProfilePair> matchGpg45Profile(
+            List<SignedJWT> credentials,
+            List<VcStoreItem> vcStoreItems,
+            Vot requestedVot,
+            AuditEventUser auditEventUser)
+            throws UnknownEvidenceTypeException, ParseException, SqsException {
+
+        Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(credentials);
+        Optional<Gpg45Profile> matchedGpg45Profile =
+                !userIdentityService.checkRequiresAdditionalEvidence(vcStoreItems)
+                        ? gpg45ProfileEvaluator.getFirstMatchingProfile(
+                                gpg45Scores, requestedVot.getValidGpg45Profiles())
+                        : Optional.empty();
+
+        // Successful match
+        if (matchedGpg45Profile.isPresent()) {
+            sendProfileMatchedAuditEvent(
+                    matchedGpg45Profile.get(), gpg45Scores, credentials, auditEventUser);
+
+            return Optional.of(new VotProfilePair(requestedVot, matchedGpg45Profile.get().label));
+        }
+        return Optional.empty();
+    }
+
+    private Optional<VotProfilePair> matchOperationalProfile(
+            List<SignedJWT> credentials, Vot requestedVot) throws ParseException {
+        for (SignedJWT cred : credentials) {
+            String credentialVot = cred.getJWTClaimsSet().getStringClaim(VOT_CLAIM);
+            Optional<String> matchedOperationalProfile =
+                    requestedVot.getValidOperationalProfiles().stream()
+                            .map(OperationalProfile::name)
+                            .filter(profileName -> profileName.equals(credentialVot))
+                            .findFirst();
+
+            // Successful match
+            if (matchedOperationalProfile.isPresent()) {
+                return Optional.of(
+                        new VotProfilePair(requestedVot, matchedOperationalProfile.get()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private List<SignedJWT> removeOperationalProfileVcs(List<SignedJWT> credentials)
```

Comment:
Looks like @rpayal has implemented a similar thing as part of his ticket as well, so we should make sure to commonise when merging the two.

I probably didn't do a good enough job of separating the implementation details for the different tickets 😬

Reply:
Dropping this here for context: https://github.com/govuk-one-login/ipv-core-back/pull/1548